analysis/topeft_run2/make_cr_and_sr_plots.py

```diff
@@ -568,29 +611,84 @@ def make_cr_fig(h_mc,h_data,unit_norm_bool,axis='process',var='lj0pt',bins=[],gr
+    err_p_stat = np.append(err_p_stat, err_p_stat[-1])                     #Check: Is it even okay to do this?
+    err_ratio_m_stat = np.append(err_ratio_m_stat, err_ratio_m_stat[-1])   #Check: Is it even okay to do this?
+    err_ratio_p_stat = np.append(err_ratio_p_stat, err_ratio_p_stat[-1])   #Check: Is it even okay to do this?
+
+    #also a good idea to keep bin edges arr handy
+    bin_edges_arr = h_mc.axes[var].edges
+
+    if unblind and plot_syst_err:
+        print("\nYou are unblinding and also plotting systematic error.")
+        err_m_syst = np.append(err_m_syst,err_m_syst[-1])
+        err_p_syst = np.append(err_p_syst,err_p_syst[-1])
+        err_ratio_m_syst = np.append(err_ratio_m_syst, err_ratio_m_syst[-1])
+        err_ratio_p_syst = np.append(err_ratio_p_syst, err_ratio_p_syst[-1])
+        bin_edges_arr = np.append(bin_edges_arr, [bin_edges_arr[-1] + (bin_edges_arr[-1] - bin_edges_arr[-2])*0.3])
+
+        #first plot the syst uncertainty band
+        ax.fill_between(bin_edges_arr,err_m_syst,err_p_syst, step='post', facecolor='none', edgecolor='gray', label='Syst err', hatch='////')
+        rax.fill_between(bin_edges_arr,err_ratio_m_syst,err_ratio_p_syst,step='post', facecolor='none', edgecolor='gray', label='Syst err', hatch='////')
+
+        #Now we calculate total uncertainty band
+        err_m_total = summed_mc_val - np.sqrt(summed_mc_sumw2+np.square(syst_err))
+        err_p_total = summed_mc_val + np.sqrt(summed_mc_sumw2+np.square(syst_err))
+        err_ratio_m_total = np.where(summed_mc_val > 0, err_m_total/summed_mc_val, 1)
+        err_ratio_p_total = np.where(summed_mc_val > 0, err_p_total/summed_mc_val, 1)
+
+        err_m_total = np.append(err_m_total, err_m_total[-1])
+        err_p_total = np.append(err_p_total, err_p_total[-1])
+        err_ratio_m_total = np.append(err_ratio_m_total, err_ratio_m_total[-1])
+        err_ratio_p_total = np.append(err_ratio_p_total, err_ratio_p_total[-1])
+
+        #Now we can plot the total uncertainty band
+        ax.fill_between(bins,err_m_total,err_p_total, step='post', facecolor='none', edgecolor='gray', label='Total err', hatch='\\\\')
+        rax.fill_between(bins,err_ratio_m_total,err_ratio_p_total,step='post', facecolor='none', edgecolor='gray', label='Total err', hatch='\\\\')
+
+    elif unblind and not plot_syst_err:
+        print("\n You are unblinding but not plotting systematic error.")
+        ax.fill_between(bins,err_m_stat,err_p_stat, step='post', facecolor='none', edgecolor='gray', label='Stat err', hatch='\\\\')
+        rax.fill_between(bins,err_ratio_m_stat,err_ratio_p_stat,step='post', facecolor='none', edgecolor='gray', label='Stat err', hatch='\\\\')
+
+    elif not unblind and plot_syst_err: #If Asimov data, no need for total error. If systematic errors available, plot that only
+        print("\nSince you are not unblinding, you only will have systematic uncertainty band in your plots")
```

Comment:
Is this because we don't actually use the MC stat errors in our analysis? If so, I'm fine with this block, but @anpicci this would probably be where you'd want to play with turning MC stat errors on/off.

Reply:
This is one of the blocks that @andrea21z or whoever else wants to use this will have to re-consider at some point. This block is not even needed, in my opinion. But I also did not want to get rid of it until we fully understand if we are correctly calculating MC stat errors in our analysis.

Reply:
Indeed, I would like to comment on this. In my opinion, we should have the option to display both systematic and statistical errors separately and in quadrature. If I understand correctly, the reason the statistical error isn't shown in the CRs is the assumption that the samples are large enough to make this error negligible. However, I'd prefer to validate this assumption and ensure that we can safely disregard MC statistical uncertainties.

Perhaps we could use the booleans `unblind`, `plot_syst_err`, and `plot_stat_err` independently of one another and avoid using pairs of them in the same if statement. This would give people full control and allow them to create combinations of boolean values depending on what they want to check.

Reply:
Ok, let's include this in the next PR.